packages/gateway/src/handlers/address.ts

```diff
@@ -10,35 +10,64 @@ interface WriteRepository {
 export function withSetAddr(repo: WriteRepository): ccip.HandlerDescription {
   return {
     type: 'setAddr',
-    func: async (args: ethers.utils.Result) => {
+    func: async (args) => {
       const params: SetAddressProps = {
         node: args.node,
         coin: args.coin,
         addr: args.addr,
       }
-      if (!params.coin) params.coin = 60 // default: ether
+      if (params.coin === undefined) params.coin = 60 // default: ether
       await repo.setAddr(params)
-      return []
+      return { data: [] }
     },
   }
 }
 
 interface ReadRepository {
-  addr(params: GetAddressProps): Promise<Response | undefined>
+  getAddr(params: GetAddressProps): Promise<Response | undefined>
 }
 
 export function withGetAddr(repo: ReadRepository): ccip.HandlerDescription {
   return {
     type: 'addr',
-    func: async (args: ethers.utils.Result) => {
+    func: async (args): Promise<ccip.HandlerResponse> => {
       const params: GetAddressProps = {
         node: args.node,
         coin: args.coin,
       }
-      if (!params.coin) params.coin = 60 // default: ether
-      const addr = await repo.addr(params)
-      if (!addr) return []
-      return [addr.value]
+      if (params.coin === undefined) params.coin = 60 // default: ether
```

Comment:
Why does the conditional changed? Also, it is possible to receive zero or something that is not a number? If so, it needs to be handle.